cmd/integration/harness.go

```diff
@@ -90,76 +77,21 @@ func NewHarness(cfg *ServerConfig, args []string) (*Harness, error) {
 	return h, nil
 }
 
-func isListening(addr string) bool {
-	conn, _ := net.DialTimeout("tcp", addr, 1*time.Second)
-	if conn != nil {
-		_ = conn.Close()
-		return true
-	}
-	return false
-}
-
-func killProcess(address string) error {
-	addr, err := net.ResolveTCPAddr("tcp", address)
-	if err != nil {
-		return err
-	}
-
-	var cmd *exec.Cmd
-	if runtime.GOOS == "windows" {
-		args := fmt.Sprintf("(Get-NetTCPConnection -LocalPort %d).OwningProcess -Force", addr.Port)
-		cmd = exec.Command("Stop-Process", "-Id", args)
-	} else {
-		args := fmt.Sprintf("lsof -i tcp:%d | grep LISTEN | awk '{print $2}' | xargs kill -9", addr.Port)
-		cmd = exec.Command("bash", "-c", args)
-	}
-
-	var errb bytes.Buffer
-	cmd.Stderr = &errb
-
-	if err := cmd.Start(); err != nil {
-		return err
-	}
-
-	if err := cmd.Wait(); err != nil {
-		return fmt.Errorf("error during killing process: %s | %s", err, errb.String())
-	}
-
-	return nil
-}
-
 func main() {
+	log.JSONLog(true)
 	// os.Args[0] contains the current process path
 	h, err := NewHarnessDefaultServerConfig(os.Args[1:])
 	if err != nil {
-		log.Error("An error has occurred while generating a new harness: %v", err)
+		log.With().Error("An error has occurred while generating a new harness: ", log.Err(err))
 	}
 	// listen on error channel, quit when process stops
-	go func() {
-		for {
-			select {
-			case errMsg := <-h.server.errChan:
-				log.Error("harness received an err from subprocess: %v", errMsg)
-			case <-h.server.quit:
-				log.Info("harness got quit signal from subprocess")
-				return
-			}
+	for {
+		select {
+		case errMsg := <-h.server.errChan:
+			log.With().Error("harness received an err from subprocess: ", log.Err(errMsg))
+		case <-h.server.quit:
+			log.With().Info("harness got a quit signal from subprocess")
```

Comment:
I don't understand what does the log mean, who creates this channel and who controls it?
is this a channel to a quit signal for the harness? or is this a quit signal coming from subprocess, and if so, when is it called? is it called on panic? these are important to prevent harness from quitting when process exits

Reply:
you are right, I added a dummy channel on the same function that receives nothing and blocking harness from terminating.

Reply:
good. please commit it :)